src/pip/_internal/metadata/_json.py

```diff
@@ -77,7 +77,7 @@ def sanitise_header(h: Union[Header, str]) -> str:
                     value = value.split()
         result[key] = value
 
-    payload = msg.get_payload()
+    payload: Union[str, None] = msg.get_payload()  # type: ignore
```

Comment:
```suggestion
    payload = cast(str, msg.get_payload())
```

Since `msg` is not multipart, we have a guarantee that the payload will be a string.